docs/source/howto/run_codes.rst

```diff
@@ -0,0 +1,372 @@
+
+If you are configuring a remote computer, start by :ref:`configuring password-less SSH access <how-to:ssh>` to it.
+
+.. note::
+
+    AiiDA will use ``bash`` on the remote computer, regardless of the default shell.
+    Please ensure that your remote ``bash`` configuration does not load a different shell.
+
+
+.. _how-to:run-codes:computer:setup:
+
+Computer setup
+--------------
+
+The configuration of computers happens in two steps: setting up the public metadata asociated with the |Computer| in AiiDA provenance graphs, and configuring private connection details.
+
+Start by creating a new computer instance in the database:
+
+.. code-block:: console
+
+   $ verdi computer setup
+
+At the end, the command will open your default editor on a file containing a summary of the configuration up to this point.
+You can add ``bash`` commands that will be executed
+
+ * *before* the actual execution of the job (under 'Pre-execution script'), and
+ * *after* the script submission (under 'Post execution script').
+
+Use these additional lines to perform any further set up of the environment on the computer, for example loading modules or exporting environment variables:
+
+.. code-block:: bash
+
+   export NEWVAR=1
+   source some/file
+
+.. note::
+
+   Don't specify settings here that are specific to a code or calculation: you can set further pre-execution commands at the ``Code`` and even ``CalcJob`` level.
+
+When you are done editing, save and quit (e.g. ``<ESC>:wq<ENTER>`` in ``vim``).
+The computer has now been created in the database but you still need to *configure* access to it using your credentials.
+
+.. tip::
+    In order to avoid having to retype the setup information the next time round, you can provide some (or all) of the information via a configuration file:
+
+    .. code-block:: console
+
+       $ verdi computer setup --config computer.yml
+
+    where ``computer.yml`` is a configuration file in the `YAML format <https://en.wikipedia.org/wiki/YAML#Syntax>`__.
+    This file contains the information in a series of key:value pairs:
+
+    .. code-block:: yaml
+
+       ---
+       label: "localhost"
+       hostname: "localhost"
+       transport: local
+       scheduler: "direct"
+       work_dir: "/home/max/.aiida_run"
+       mpirun_command: "mpirun -np {tot_num_mpiprocs}"
+       mpiprocs_per_machine: "2"
+       prepend_text: |
+          module load mymodule
+          export NEWVAR=1
+
+   The list of the keys for the ``yaml`` file is given by the options of the ``computer setup`` command:
+
+   .. code-block:: console
+
+      $ verdi computer setup --help
+
+   Note: Remove the ``--`` prefix and replace ``-`` within the keys by the underscore ``_``.
+
+.. _how-to:run-codes:computer:configuration:
+
+Computer configuration
+----------------------
+
+The second step configures private connection details using:
+
+.. code-block:: console
+
+   $ verdi computer configure TRANSPORTTYPE COMPUTERNAME
```

Comment:
```suggestion
   $ verdi computer configure TRANSPORTTYPE COMPUTERLABEL
```

Reply:
Just for me to know: is computer "name" finally deprecated?
If yes, I'll replace the other instances of it in the docs/code...

Reply:
Yeah, to my surprise it is still there for `Computer`, however, for `Code` everything has already been renamed from `name` to `label` for consistency. We should do the same for computers. Found the issue: #2926  I brought this up during the alpha and beta versions of 1.0 and we decided to push this to 2.0. So I guess it would still be okay to use name for computers, but since the CLI uses label (so the config file also has `label` as a key) I think it would already be better to start moving this to label.